python/amici/sbml_import.py

```diff
@@ -742,16 +763,52 @@ def _process_rules(self) -> None:
             _check_unsupported_functions(formula, 'Rule')
 
             if variable in stoichvars:
-                self.stoichiometric_matrix = \
-                    self.stoichiometric_matrix.subs(variable, formula)
+                if variable not in self.compartment_symbols: # change to only self.compartment_rules symbols?
+                    self.stoichiometric_matrix = \
+                        self.stoichiometric_matrix.subs(variable, formula)
+                #else:
+                #    self.stoichiometric_matrix = \
+                #        self.stoichiometric_matrix.subs(variable, self.compartment_volume[list(self.compartment_symbols).index(variable)])
 
             if variable in specvars:
                 raise SBMLException('Species assignment rules are currently'
                                     ' not supported!')
 
             if variable in compartmentvars:
-                raise SBMLException('Compartment assignment rules are'
-                                    ' currently not supported!')
+                if rule.getTypeCode() == sbml.SBML_ASSIGNMENT_RULE:
+                    self.compartment_rules[variable] = {
+                            'type_code': sbml.SBML_ASSIGNMENT_RULE,
+                            'formula': formula,
+                            # setting v0 might be unnecessary. currently in place to satisfy the ODEModel _equation_from_component method (which I edited to require v0...)
+                            'v0': self.compartment_volume[list(self.compartment_symbols).index(variable)].subs(
+                                self.symbols['species']['identifier'],
+                                self.symbols['species']['value']
+                            )
+                        }
+                    self.compartment_volume[list(self.compartment_symbols).index(variable)] = formula
+                    assignments[str(variable)] = formula
+                    #continue # avoid simplifying substitution of the compartment expression, to avoid possibly circular substitution
+                elif rule.getTypeCode() == sbml.SBML_RATE_RULE:
+                    self.compartment_rules[variable] = {
+                            'type_code': sbml.SBML_RATE_RULE,
+                            'formula': formula,
+                            'v0': self.compartment_volume[list(self.compartment_symbols).index(variable)]
+                        }
+
+                    self.symbols['species']['identifier'] = self.symbols['species']['identifier'].col_join(sp.Matrix([variable]))
```

Comment:
Would it be possible to directly implement this in `_process_species`? Would have the benefit that we won't have to worry about downstream effects later on.